lightning/src/chain/keysinterface.rs

```diff
@@ -245,12 +247,55 @@ pub trait ChannelKeys : Send+Clone {
 	/// return value must contain a signature.
 	fn sign_local_commitment_htlc_transactions<T: secp256k1::Signing + secp256k1::Verification>(&self, local_commitment_tx: &LocalCommitmentTransaction, local_csv: u16, secp_ctx: &Secp256k1<T>) -> Result<Vec<Option<Signature>>, ()>;
 
+	/// Create a signature for a justice transaction spending a revokable output on a revoked
+	/// remote commitment or HTLC transaction.
+	///
+	/// Justice transaction may claim multiples outputs at same time if timelock are similar.
+	/// It may be called multiples time for same output(s) if a fee-bump is needed with regards
+	/// to an upcoming timelock expiration.
+	///
+	/// Witness_script is a revokable witness script as defined in BOLT3 for `to_local`/HTLC
+	/// outputs.
+	///
+	/// Amount is value of the output spent by this input.
+	///
+	/// Per_commitment key is revocation secret such as provided by remote party while
```

Comment:
Maybe lets call it per_commitment secret in our public stuff. Its not a "key" in the sense that we never use it for signing, its just a secret value which is then used to derive signing keys.